src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AsyncInstanceCreationManager.java

```diff
@@ -0,0 +1,60 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.util;
+
+import cloud.fogbow.ras.constants.Messages;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+This class helps to cope Plugin that it has asynchronous instances creation.
+
+Problem: It might generate resource trash in the cloud due to the fact that
+If the RAS shutdown this class will lost its data in memory.
+
+Note: This context helps to fix this issue: https://github.com/fogbow/resource-allocation-service/issues/435
+ */
+public class AsyncInstanceCreationManager {
+
+    private static final Logger LOGGER = Logger.getLogger(AsyncInstanceCreationManager.class);
+
+    private final static List<String> creating = new ArrayList<>();
+
+    /*
+    It must be used soon before the plugin makes asynchronous creation operation in the cloud.
+
+    @return It is a callback that it must be used when the cloud finish asynchronous creation operation in the cloud;
+    Suggestion: Use "finishCreationCallback" as attribute name.
+       */
+    // TODO(chico) - Add log debug
+    public Runnable startCreation(String instanceId) {
+        LOGGER.debug(Messages.Info.START_ASYNC_INSTANCE_CREATION_S);
+        defineAsCreating(instanceId);
+        return () -> {
+            defineAsCreated(instanceId);
+            LOGGER.debug(Messages.Info.END_ASYNC_INSTANCE_CREATION_S);
```

Comment:
Format the message info string to fill the %s variable in the string.

Reply:
Done